### ISSUE TITLE
Creating a new record error

### ISSUE THREAD

**Issue by KrisLau** [Nozbe/WatermelonDB#1153]
I've been following these two WatermelonDB documents:
- https://nozbe.github.io/WatermelonDB/Writers.html
- https://nozbe.github.io/WatermelonDB/CRUD.html
I might've missed something but it doesn't seem to explain how to call the writer method that we write? I tried to figure it out by myself and tried implementing it . However I've been getting this error message:
```
Unhandled promise rejection [TypeError: undefined is not a function (near '...userCollection.addUser...')]
```

In my User model, I have defined this writer method and custom query:
```
@lazy
  profiles = this.collections
    .get('profile')
    .query(Q.on('profile_user', 'user_id', this.id));

  @lazy
  userTeams = this.profiles
    .get('team')
    .query(Q.on('team_member', 'profile_id', this.id));

@writer async addUser(userID, firstName, lastName, email, phoneNumber) {
    await this.collection.create(user => {
        user.userID = userID;
        user.firstName = firstName;
        user.lastName= lastName;
        user.email= email;
        user.phoneNumber = phoneNumber;
    });
}
```

Then in my authentication service, I'm trying to add the user into the `user` table when they sign in for the first time so I did: 
```
const database = useDatabase();
const userCollection = database.get('user')
.
.
.
// after sign in resolves, add user to watermelonDB user table
auth()
        .signInWithCredential(googleCredential)
        .then(async user => {
          if (user.additionalUserInfo.isNewUser) {
            await userCollection.addUser(
              auth().currentUser.uid,
              user.additionalUserInfo.profile.given_name,
              user.additionalUserInfo.profile.family_name,
              auth().currentUser.email,
              null,
            );
          }
        });
```

Other things I tried: 
- `database.addUser`
- Trying to do a database `get` then calling the writer method on the `get` result
```
auth()
        .signInWithCredential(googleCredential)
        .then(async user => {
          const userCollection = database.get('user');
          if (user.additionalUserInfo.isNewUser) {
            await userCollection.addUser(
              auth().currentUser.uid,
              user.additionalUserInfo.profile.given_name,
              user.additionalUserInfo.profile.family_name,
              auth().currentUser.email,
              null,
            );
          }
        });
```

In my home screen, I can't seem to figure out how to call the custom query and I'm not even sure that i defined them correctly. A user can have many profiles and a profile can be a part of a team. I'm trying to get all the user's teams but I'm struggling to figure out how to get that

**TL;DR: How do I call the custom query and writer method?**

Amy help with this would be greatly appreciated!

**Comment by KrisLau** [comment]
I have also tried changing `this.collection.create` to `this.collection.table.create`  and `this.create` but it still didn't work. I also tried changing the way I called the `addUser` but that was also not fruitful. 

Looking through the example, it looks like it still uses the `@action` decorator instead of the `@writer` decorator. So I was not able to find any external tutorials in order to get it to work.

**Comment by KrisLau** [comment]
Trying to use `withObservables` on my screen to get the teams for a user but it's still not working:
```

```

**Comment by justinairbase** [comment]
@KrisLau Hey, how did you manage to trigger the writer?

**Comment by KrisLau** [comment]
@justinairbase I pretty much gave up on it cause it was wasting too much time and I'm on a tight schedule hahha

**Comment by Garabed96** [comment]
@KrisLau did you ever go back and create a working version with in Model writers, I've seen absolutely no examples online.
